pandas/tests/io/json/test_normalize.py

```diff
@@ -463,6 +463,19 @@ def test_nested_flattening_consistent(self):
         # They should be the same.
         tm.assert_frame_equal(df1, df2)
 
+    def test_nonetype_record_path(self, nulls_fixture):
+        # see gh-30148
+        # should not raise TypeError
+        df1 = json_normalize(
+            [
+                {"state": "Texas", "info": nulls_fixture},
+                {"state": "Florida", "info": [{"i": 2}]},
+            ],
+            record_path=["info"],
+        )
+        df2 = DataFrame({"i": 2}, index=[0])
```

Comment:
use result= and expected =
tm.assert_equal(result, expected)

Reply:
Done. This is different btw from the test above this test.